spec/models/import/transactions/actual_and_refund_csv_row_spec.rb

```diff
@@ -0,0 +1,583 @@
+            allow(csv_row).to receive(:field).with("Receiving Organisation Type").and_return("60")
+            row = described_class.new(csv_row)
+
+            expect(row).to be_valid
+        end
+      end
+      context "when the value is not on the code list" do
+        it "is invalid with an appropriate error" do
+            csv_row = valid_csv_row
+            allow(csv_row).to receive(:field).and_call_original
+            allow(csv_row).to receive(:field).with("Receiving Organisation Type").and_return("not a code")
+            row = described_class.new(csv_row)
+
+            expect(row).to be_invalid
+            expect(row.errors["Receiving Organisation Type"][1]).to include("valid IATI Organisation Type code")
+        end
+      end
+    end
+
+    describe "activity comments" do
+      context "when the row contains a valid activity comment" do
+        it "is valid" do
+            csv_row = valid_csv_row
+            allow(csv_row).to receive(:field).and_call_original
+            allow(csv_row).to receive(:field).with("Actual Value").and_return("0.00")
+            allow(csv_row).to receive(:field).with("Refund Value").and_return("0.00")
+            allow(csv_row).to receive(:field).with("Comment").and_return("This is a valid activity comment")
+            row = described_class.new(csv_row)
+
+            expect(row).to be_valid
+        end
+      end
+
+      context "When the actual and refund values are blank with a comment" do
+        it "is invalid with an appropriate message" do
+            csv_row = valid_csv_row
+            allow(csv_row).to receive(:field).and_call_original
+            allow(csv_row).to receive(:field).with("Actual Value").and_return(nil)
+            allow(csv_row).to receive(:field).with("Refund Value").and_return(nil)
+            allow(csv_row).to receive(:field).with("Comment").and_return("This is a valid activity comment")
+            row = described_class.new(csv_row)
+
+            expect(row).to be_invalid
+            expect(row.errors["Comment"][1]).to include("must both be zero to provide a comment")
+        end
+      end
+
+      context "when the actual and refund values are zero without a comment" do
+        it "is invalid with an appropriate message" do
+            csv_row = valid_csv_row
+            allow(csv_row).to receive(:field).and_call_original
+            allow(csv_row).to receive(:field).with("Actual Value").and_return("0.00")
+            allow(csv_row).to receive(:field).with("Refund Value").and_return("0.00")
+            allow(csv_row).to receive(:field).with("Comment").and_return(nil)
+            row = described_class.new(csv_row)
+
+            expect(row).to be_invalid
+            expect(row.errors["Comment"][1]).to include("with 0 actual and 0 refund require a comment")
+        end
+      end
+
+      context "when all of the required values are blank" do
+        it "is invalid with an appropriate message" do
```

Comment:
The description says invalid, but the assertion on row 557 says valid, so I'm confused!